genesyscloud/telephony_provider_edges_trunkbasesettings/genesyscloud_telephony_providers_trunkbasesettings_proxy.go

```diff
@@ -0,0 +1,196 @@
+				}
+			}
+
+			if name == "" {
+				break
+			}
+
+		}
+		return nil
+	})
+
+	return &trunkbaseSlice, nil, nil
+}
+
+// The SDK function is too cumbersome because of the various boolean query parameters.
+// This function was written in order to leave them out and make a single API call
+func getTelephonyProvidersEdgesTrunkbasesettings(p *trunkbaseSettingProxy, pageNumber int, pageSize int, name string) (*platformclientv2.Trunkbaseentitylisting, *platformclientv2.APIResponse, error) {
+	headerParams := make(map[string]string)
+	if p.clientConfig.AccessToken != "" {
+		headerParams["Authorization"] = "Bearer " + p.clientConfig.AccessToken
+	}
+	// add default headers if any
+	for key := range p.clientConfig.DefaultHeader {
+		headerParams[key] = p.clientConfig.DefaultHeader[key]
+	}
+
+	queryParams := make(map[string]string)
+	queryParams["pageNumber"] = p.clientConfig.APIClient.ParameterToString(pageNumber, "")
+	queryParams["pageSize"] = p.clientConfig.APIClient.ParameterToString(pageSize, "")
+	if name != "" {
+		queryParams["name"] = p.clientConfig.APIClient.ParameterToString(name, "")
+	}
+
+	// to determine the Content-Type header
+	httpContentTypes := []string{"application/json"}
+
+	// set Content-Type header
+	httpContentType := p.clientConfig.APIClient.SelectHeaderContentType(httpContentTypes)
+	if httpContentType != "" {
+		headerParams["Content-Type"] = httpContentType
+	}
+
+	// set Accept header
+	httpHeaderAccept := p.clientConfig.APIClient.SelectHeaderAccept([]string{
+		"application/json",
+	})
+	if httpHeaderAccept != "" {
+		headerParams["Accept"] = httpHeaderAccept
+	}
+	var successPayload *platformclientv2.Trunkbaseentitylisting
+	path := p.clientConfig.BasePath + "/api/v2/telephony/providers/edges/trunkbasesettings"
+	response, err := p.clientConfig.APIClient.CallAPI(path, http.MethodGet, nil, headerParams, queryParams, nil, "", nil, "")
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if response.Error != nil {
+		err = errors.New(response.ErrorMessage)
+	} else {
+		err = json.Unmarshal(response.RawBody, &successPayload)
+	}
+
+	return successPayload, response, err
+}
+
+func createTrunkBaseSettingFn(ctx context.Context, p *trunkbaseSettingProxy, trunkBaseSetting platformclientv2.Trunkbase) (*platformclientv2.Trunkbase, *platformclientv2.APIResponse, error) {
+	return p.edgesApi.PostTelephonyProvidersEdgesTrunkbasesettings(trunkBaseSetting)
+}
+
+func updateTrunkBaseSettingFn(ctx context.Context, p *trunkbaseSettingProxy, trunkbaseSettingId string, trunkBaseSetting platformclientv2.Trunkbase) (*platformclientv2.Trunkbase, *platformclientv2.APIResponse, error) {
+	return p.edgesApi.PutTelephonyProvidersEdgesTrunkbasesetting(trunkbaseSettingId, trunkBaseSetting)
+}
+
+func deleteTrunkBaseSettingFn(ctx context.Context, p *trunkbaseSettingProxy, trunkBaseSettingId string) (*platformclientv2.APIResponse, error) {
+	return p.edgesApi.DeleteTelephonyProvidersEdgesTrunkbasesetting(trunkBaseSettingId)
+}
```

Comment:
I think we should delete the cache item here after a successful delete via API. Here's an example of that being done elsewhere - https://github.com/MyPureCloud/terraform-provider-genesyscloud/blob/main/genesyscloud/routing_queue/genesyscloud_routing_queue_proxy.go#L227

If the exporter is active at the same time as a delete, then the delete function will try to read the object until it sees a 404, but it will never see this since it will be reading from the cache map.

I think this can only be an issue in our test cases and not while actually using terraform, but still better to avoid potential test failures in the future.